src/orion/core/io/config.py

```diff
@@ -451,3 +451,35 @@ def to_dict(self):
                 config[key] = self[key].to_dict()
 
         return config
+
+    def from_dict(self, config):
```

Comment:
I think this new method would deserve a couple of tests in `tests/unittests/core/io/test_config.py`. Among other things, what if the config contains additional keys? What if it lacks some keys? What if a value is not valid? What if it's a deprecated key in the config? Are subconfiguration handled properly? Even values that are dict?

Reply:
* what if the config contains additional keys

we iterate over the _config keys not the provided config so those are ignored

* What if it lacks some keys

we iterate over the `_config` keys  not `config` so we iterate over everything that is defined and pop what is not

* What if a value is not valid

it uses ``Configuration.__setitem__`` so the value gets validated and the expected TypeError exception is raised

* Are subconfiguration handled properly?

Yes, storage has the Database subconfig.

Reply:
Thanks for the answers, I could see it from the code. :grinning: These questions were meant as potential test-case for the unittests.

Reply:
I am going to add the tests

Reply:
Tests added